providers/src/airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -620,6 +625,9 @@ def execute_sync(self, context: Context):
                 self.callbacks.on_pod_creation(
                     pod=self.remote_pod, client=self.client, mode=ExecutionMode.SYNC
                 )
+
+            self.await_init_containers_completion(pod=self.pod)
+
```

Comment:
Before we start reading the logs from the main containers, we wait for the pod to start and we need to do the same before reading the logs from the init containers, otherwise, the reading will fail if the pod is not scheduled yet.

If I'm not mistaken, when the init containers start, the pod status will be `Pending`, but the pod should have a condition `Initialized` set to false (any state after that should be ok, for ex `Initialized`, `Running`, ... )

Reply:
Look like we can use `Waiting` instead. While init_container `Waiting` we should wait. 

> If a container is not in either the Running or Terminated state, it is Waiting. A container in the Waiting state is still running the operations it requires in order to complete start up: for example, pulling the container image from a container image registry, or applying [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) data. When you use kubectl to query a Pod with a container that is Waiting, you also see a Reason field to summarize why the container is in that state.

ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-state-waiting

So we can have `self._await_init_container_start(pod=pod, container_name=c)` that will wait non-empty non-wait status :
```
        # sort by spec.initContainers because containers runs sequentially
        containers_to_log = sorted(containers_to_log, key=lambda cn: all_containers.index(cn))
        for c in containers_to_log:
            self._await_init_container_start(pod=pod, container_name=c)
            status = self.fetch_container_logs(pod=pod, container_name=c, follow=follow_logs)
            pod_logging_statuses.append(status)
```

Reply:
Updated using `remote_pod.status.phase` that should not be equal to `PodPhase.PENDING`

Reply:
any more comments @hussein-awala ? I am not too fluent in that part, but maybe we can get this to completion?